src/Fields/abstract_field.jl

```diff
@@ -27,23 +26,16 @@ Abstract supertype for fields located at `(X, Y, Z)` on architecture `A`
 and defined on a grid `G` with eltype `T` and `N` dimensions.
 """
 abstract type AbstractField{X, Y, Z, A <: ArchOrNothing, G <: GridOrNothing, T, N} <: AbstractArray{T, N} end
```

Comment:
That's a good question.

It's useful here:

https://github.com/CliMA/Oceananigans.jl/blob/d5f61a089876853ab3122707938b0d7727c12e61/src/Fields/set!.jl#L21-L43

but the utility is pretty minor.

It also requires us to use the child architecture for that type parameter in distributed fields:

https://github.com/CliMA/Oceananigans.jl/blob/d5f61a089876853ab3122707938b0d7727c12e61/src/Distributed/distributed_fields.jl#L7

I kept the `A` type parameter here to reduce how much refactoring I had to do (since it was already enormous).

But I do think there are still some improvements to make...

Reply:
I've decided to bite the bullet...